code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,274 @@
+	name = "paper= 'A Study in Terror'"
+	info = "<b>Findings Overview</b><br>\
+	<br>\
+	<p>Gray Spiders<br>Stealthy predators that ambush prey from vents. Otherwise weak.</p>\
+	<p>Red Spiders<br>Can kill anyone in melee, but slow.</p>\
+	<p>Green<br>Will lay eggs on dead bodies, breeding more spiders.</p>\
+	<p>Black<br>Even a single bite is enough to kill a humanoid, given time.</p>\
+	<p>White<br>Injects a parasitic agent. Deemed to pose too great an infection risk to study.</p>\
+	<p>Purple<br>Only seen guarding the nest of the Queen to the south. Appear to be territorial, and very dangerous.</p>\
+	<p>Prince<br>Held in containment 2. Appears to be a sort of super-warrior. Fast, strong, and thickly armored.</p>\
+	<p>Mother<br>Carries hordes of spiderlings on its back. Held in containment 1. </p>\
+	<p>Queen<br>Unable to contain. Present south of Cargo before contact was lost. Presumed ruler of the local hive.</p>\
+	"
+
+/obj/item/weapon/paper/terrorspiders6
+	name = "paper= 'A Study in Venom'"
+	info = "<b>Initial Report - Black Widow Venom</b><br>\
+	<br>\
+	Even a single bite kills within a few minutes. Immediate charcoal is necessary for the victim to survive.<br>\
+	The fascinating thing is that multiple bites actually cause rapid symptom onset. The victim quickly loses the ability to walk.<br>\
+	One test monkey went from perfect health to death in under 30 seconds - from 3 bites.<br>\
+	I am so very glad I'm not responsible for keeping these things contained."
+
+/obj/item/weapon/paper/terrorspiders7
+	name = "paper= 'Security Orders'"
+	info = "<b>Directives for Officer James</b><br>\
+	<br>\
+	1. Find a way to lift the bridge lockdown.<br>\
+	2. Kill any spiders you encounter.<br>\
+	3. Once on the bridge, open a path to the Queen in the south, and find a way to kill her, ending the infestation. <br>\
+	Good luck, Officer.<br>\
+	"
+
+/obj/item/weapon/paper/terrorspiders8
+	name = "paper= 'Last Words'"
+	info = "<b>The Last Words of Nurse Barnes</b><br>\
+	<br>\
+	It is too much. First Phil, then those damn spiders bite Mary too. Heaven help us. <br>\
+	I'll be evacuating as soon as I've grabbed my supplies. I had to pay for those myself. Stupid budget cuts. <br>\
+	The new world they've found for us isn't that hospitable, but at least there are no deadly spid---<br>\
+	(the writing trails off, as if the writer was interrupted)<br>\
+	"
+
+
+/obj/item/weapon/gun/energy/laser/awaymission_aeg
+	name = "Wireless Energy Gun"
+	desc = "An energy gun that recharges wirelessly during away missions. Does not work on the main station."
+	force = 10
+	origin_tech = null
+	selfcharge = 1
+	var/inawaymission = 1
+
+/obj/item/weapon/gun/energy/laser/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == (MAX_Z + 1))
+		if(inawaymission)
+			..()
+		else
+			to_chat(get(src, /mob), "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
```

Comment:
just do
```
if(ismob(loc))
    to_chat(loc, "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
```
it's consistent with how everything else does it and the user doesn't need notifications if they have it shoved in their backpack anyways